src/overlays/actors/ovl_Obj_Kendo_Kanban/z_obj_kendo_kanban.c

```diff
@@ -217,238 +233,251 @@ void ObjKendoKanban_Destroy(Actor* thisx, PlayState* play) {
         Matrix_Pop();
     }
 
     Actor_UpdateBgCheckInfo(play, &this->actor, 0.0f, 0.0f, 0.0f, 4);
 
     if (this->actor.bgCheckFlags & 1) {
+        // When on the ground, apply some friction.
         this->actor.velocity.x *= 0.8f;
         this->actor.velocity.z *= 0.8f;
     }
 
-    if (this->unk_300 == 1) {
-        if (this->unk_308 >= 7) {
-            s16 temp_v0_3 = this->unk_302 & 0x3FFF;
-
-            if (temp_v0_3 >= 0x2000) {
-                temp_v0_3 -= 0x4000;
+    if (this->bHasNewRootCornerPos == true) {
+        if (this->numBounces >= MAX_BOUNCE_COUNT) {
+            s16 deltaRotationAngle = this->rotationAngle & 0x3FFF; // 90 degrees
+            if (deltaRotationAngle >= 0x2000) {                    // 45 degrees
+                deltaRotationAngle -= 0x4000;                      // 90 degrees
             }
-            this->unk_302 -= temp_v0_3;
-            this->unk_304 = 0;
-            func_80B65D54(this);
+            this->rotationAngle -= deltaRotationAngle;
+            this->rotationVelocity = 0;
+            ObjKendoKanban_SetupSettled(this);
             return;
         }
 
         if (this->actor.bgCheckFlags & 2) {
+            // Upon touching the ground...
             Actor_PlaySfxAtPos(&this->actor, NA_SE_EV_WOODPLATE_BOUND);
-            this->unk_300 = 0;
+            this->bHasNewRootCornerPos = false;
             this->actor.velocity.y *= 0.5f;
+
         } else if (this->actor.bgCheckFlags & 1) {
-            this->unk_308++;
-            this->unk_300 = 0;
+            // When on the ground...
+            this->numBounces++;
+            this->bHasNewRootCornerPos = false;
             this->actor.velocity.x *= 0.3f;
             this->actor.velocity.z *= 0.3f;
             Actor_PlaySfxAtPos(&this->actor, NA_SE_EV_WOODPLATE_BOUND);
 
-            if (sp38 > 0.0f) {
-                if (this->unk_304 > 0) {
-                    this->unk_304 *= 1.2f;
+            // Adjust and (potentially) reverse rotation depending on the current
+            // facing of the board and the direction in which it is rotating.
+            if (verticalScalar > 0.0f) {
+                if (this->rotationVelocity > 0) {
+                    this->rotationVelocity *= 1.2f;
                 } else {
-                    this->unk_304 *= -0.6f;
+                    this->rotationVelocity *= -0.6f;
                 }
-            } else if (this->unk_304 < 0) {
-                this->unk_304 *= 1.2f;
             } else {
-                this->unk_304 *= -0.6f;
+                if (this->rotationVelocity < 0) {
+                    this->rotationVelocity *= 1.2f;
+                } else {
+                    this->rotationVelocity *= -0.6f;
+                }
             }
         }
     }
 }
 
-s32 func_80B6618C(ObjKendoKanban* this, PlayState* play) {
+s32 ObjKendoKanban_IsPlayerOnTop(ObjKendoKanban* this, PlayState* play) {
+    const s32 UNDETERMINED = 0;
+    const s32 UP = 1;
+    const s32 DOWN = -1;
```

Comment:
This would probably be better as an enum, rather than three distinct constants.

Reply:
Agreed.